src/angular-pdfjs-viewer.js

```diff
@@ -172,7 +172,7 @@
                 });
 
                 // watch other attributes
-                $scope.$watch(function () {
+                $element.on('$destroy', $scope.$watch(function () {
```

Comment:
Great catch - did it help?

Reply:
YES, looks pretty stable, couldn't reproduce the issue after navigating over 50 candidates